src/index.tsx

```diff
@@ -209,10 +211,24 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => {
+                if(onBeforeGetContent) {
+                    const onBeforeGetContentOutput = onBeforeGetContent();
+                    if(onBeforeGetContentOutput && typeof onBeforeGetContentOutput.then === "function") {
+                        onBeforeGetContentOutput.then(() => {
+                        this.handlePrint();
+                        })
+                    } else {
+                        return this.handlePrint();
+                    }
+                } else {
+                     return this.handlePrint();
```

Comment:
Nit: this return is no longer needed

Reply:
I think we need this return because if the user doesn't pass onBeforeGetContentOutput method, we will return the handlePrint method to onClick. Remember that we need to return since our function is inside **{}**

Reply:
See my comment above

Reply:
changed